spec/tests/helpers/perf_run_helper_spec.rb

```diff
@@ -439,32 +439,63 @@ class PerfRunHelperClass
     end
   end
   describe "#copy_system_logs" do
-    let!(:hosts) do
-      [Beaker::Host.create("master",
-                           { platform: Beaker::Platform.new("centos-6.5-x86_64"),
-                             role: "master" }, logger: @logger)]
+    let(:host) { double.as_null_object }
+    let(:archive_root) { "/tmp/test/archive" }
+    let(:archive_dir) { "/var/log" }
+    let(:archive_name) { "puppet_logdir.tgz" }
+    let(:archive_path) { File.join(archive_dir, archive_name) }
+
+    context "when archiving succeeds" do
+      it "calls scp_from to copy archive path to archive root" do
+        expected_dest = File.join(archive_root, host)
+
+        # mock out @archive_root instance variable
+        subject.instance_variable_set(:@archive_root, archive_root)
+
+        allow(subject).to receive(:archive_system_logs).with(host).and_return(archive_path)
+        expect(subject).to receive(:scp_from).with(host, archive_path, expected_dest)
+
+        subject.copy_system_logs(host)
+      end
     end
-    it "calls archive_file_from with given host and parent dir of puppet_logdir" do
-      # test variables
-      build_id = gplt_timestamp = nil
-      puppet_logdir = "/foo/bar/baz"
-      archive_root = "/test/archive"
-      job_name = "foobar"
-      archive_name = "#{job_name}__#{build_id}__#{gplt_timestamp}__system_logs.tgz"
-      ENV["JOB_NAME"] = job_name
-      myhost = hosts.first
-      # mock out @archive_root instance variable
-      subject.instance_variable_set(:@archive_root, archive_root)
-      # mock out beaker result
-      result = Beaker::Result.new(myhost, "foo")
-      result.stdout = puppet_logdir
-
-      allow(subject).to receive(:on).with(myhost, any_args).and_return(result)
-      expect(subject).to receive(:puppet).with("config", "print", "logdir")
-      expect(subject).to receive(:archive_file_from).with(myhost,
-                                                          File.dirname(puppet_logdir),
-                                                          {}, archive_root, archive_name)
-      subject.copy_system_logs(myhost)
+    context "when archiving fails" do
+      it "scp_from is not called" do
+        allow(subject).to receive(:archive_system_logs).with(host).and_return(nil)
+        expect(subject).not_to receive(:scp_from)
```

Comment:
See comment above; this example should probably expect not to receive the `FileUtils.mkdir_p` as well.

Reply:
Thanks @billclaytor I have added a test for the anti mkdir_p expectation.